third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/DotOpToLLVM/MMAv2.cpp

```diff
@@ -330,9 +376,13 @@ LogicalResult convertDot(const LLVMTypeConverter *typeConverter,
   int repM = repA[1], repN = repB[2], repK = repA[2];
   int repBatch = repA[0];
 
-  // shape / shape_per_cta
   auto ha = getValuesFromDotOperandLayoutStruct(
       typeConverter, loc, rewriter, loadedA, repBatch, repM, repK, aTensorTy);
+
+  // FIXME [Dot LL]
+  // max(repN / 2, 1) is wrong for repN = 1!
```

Comment:
Can you elaborate on `// max(repN / 2, 1) is wrong for repN = 1!`?
Why `repN=1` is wrong?

Reply:
We are taking this `max(repN / 2, 1)` here, and then in the loop inside `getValuesFromDotOperandLayoutStruct` we are packing 4 elements at a time. Rather than that, the correct implementation packs 2 elements inside the function for `opIdx=1` and iterates `repN` times.

Reply:
Got it